### PR TITLE
Output format improvements

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vltrace/22)

### CODE REVIEW

File: test/helper_functions.sh [hunk 1]
```diff
@@ -104,16 +104,14 @@ function split_forked_file() {
...
 	while true; do
 		NAME="${NAME1}-${N}-${NAME2}"
 		touch $NAME
 		set +e
-		[ "$PID" == "fork" ] \
-			&& grep "$PID" $INPUT | sort -k3 -r > $NAME \
-			|| grep "$PID" $INPUT > $NAME
-		grep -v "$PID" $INPUT > $GREP
+		grep -e "$PID" $INPUT > $NAME
+		grep -v -e "$PID" $INPUT > $GREP
```
Comment (marcinslusarz) [inline test/helper_functions.sh:114]: This seems to be unrelated change. Can you split it to separate commit and describe what and why you do that?

**Review by marcinslusarz — CHANGES_REQUESTED** [review]

File: test/helper_functions.sh (hunk 1 above)
Comment (ldorau) [reply test/helper_functions.sh:114]: Done

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 51 of 51 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/22#-:-Kq7mL4MNGj1OcwmXAfV:bnfp4nl)*

**Review by marcinslusarz — APPROVED** [review]

File: src/ebpf_syscalls.h [hunk 2]
```diff
@@ -141,8 +141,8 @@ enum {
 	SC_NAME_LEN = 31
 };
 
-/* properties of syscall with number 'num' */
-struct syscall_descriptor {
+/* properties of syscall with number 'num' (syscall descriptor) */
+struct sc_desc {
```
Comment (GBuella) [inline src/ebpf_syscalls.h:146]: The original type name was just too long?

File: src/print_event_cb.c [hunk 3]
```diff
@@ -259,7 +261,11 @@ fprint_path(unsigned path, int *str_fini, FILE *f,
 		*str_fini = 1;
 	}
 
+	if (!event->info.is_cont)
+		fwrite("\"", 1, 1, f);
```
Comment (GBuella) [inline src/print_event_cb.c]: or `fputc('\"', f);`

File: src/print_event_cb.c [hunk 4]
```diff
@@ -296,11 +302,11 @@ fwrite_out_lf_fld_sep(FILE *f)
...
 {
-	static char str[] = "_----------------_----------------_";
+	static char str[] = "_------------------_------------------_";
+	char *c;
 
-	str[0]  = Args.separator;
-	str[17] = Args.separator;
-	str[34] = Args.separator;
+	while (c = strchr(str, '_'))
+		*c = Args.separator;
```
Comment (GBuella) [inline src/print_event_cb.c:309]: Wait, what?
Modifying a string literal?
That is blasphemy!
Also undefined behavior.
Also, compiler can ask the linker to place these string literals in the rodata segment, which is going to be mapped as read-only memory pages - resulting in segfault.

I guess something like this would make me able to sleep at night:
```
static const char str_template[] = "_------------------_------------------_";
static char str_store[sizeof(str_template)];
strcpy(str_store, str_template);
```

Comment (GBuella) [reply src/print_event_cb.c:309]: I know this wasn't introduced with this PR, but I noticed it just now.

**Review by GBuella — CHANGES_REQUESTED** [review]

File: src/print_event_cb.c (hunk 4 above)
Comment (GBuella) [reply src/print_event_cb.c:309]: Oh wait, this `str` might not count as something holding a string literal, gotta look into it!

Comment (GBuella) [reply src/print_event_cb.c:309]: Oh, got it, `str[]` here is of course not a string literal, but a char array initialized using a string literal. So it should end up in the good old data segment.
I hope.
ah, I fooled myself!

File: src/ebpf_syscalls.h (hunk 2 above)
Comment (ldorau) [reply src/ebpf_syscalls.h:146]: Yes

File: src/print_event_cb.c (hunk 4 above)
Comment (ldorau) [reply src/print_event_cb.c:309]: That' right - ```str[]``` here is a char array initialized using a string literal

File: src/print_event_cb.c (hunk 3 above)
Comment (ldorau) [reply src/print_event_cb.c]: Right, changed.

**Comment by ldorau** [comment]
Rebased on master

**Review by GBuella — APPROVED** [review]

**Comment by marcinslusarz** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/22)*
